roundabout/utils.js

```diff
@@ -3,7 +3,74 @@ export const MAX_EXPIRES_IN = 3 * 24 * 60 * 60 // 7 days in seconds
 export const MIN_EXPIRES_IN = 1
 export const DEFAULT_EXPIRES_IN = 3 * 24 * 60 * 60 // 3 days in seconds by default
 
-export const VALID_BUCKETS = ['dagcargo']
+export const VALID_BUCKETS_BY_KEY = ['dagcargo']
+export const VALID_R2_BUCKETS_DEFAULT = ['carpark-prod-0', 'carpark-prod-1', 'dagcargo']
+export const VALID_S3_BUCKETS_DEFAULT = ['carpark-prod-0']
+export const CF_R2_DOMAIN = 'r2.cloudflarestorage.com'
+export const AWS_S3_DOMAIN = 's3.amazonaws.com'
+
+/**
+ * Filters location claims to get the R2 buckets valid for redirect.
+ * In case of not existing any R2 bucket, verifies if there is an AWS bucket name that could be attempted in CF.
+ * 
+ * @param {Set<string>} locations 
+ * @param {object} [options]
+ * @param {string[]} [options.validR2Buckets]
+ * @param {string[]} [options.validS3Buckets]
+ */
+export function getBucketKeyPairToRedirect (locations, options = {}) {
+  const validR2Buckets = options.validR2Buckets || VALID_R2_BUCKETS_DEFAULT
+  const validS3Buckets = options.validS3Buckets || VALID_S3_BUCKETS_DEFAULT
+  // Filter by Cloudflare R2 URLs
+  const r2Urls = Array.from(locations)
+    .filter(
+      // CF Domain
+      l => l.includes(CF_R2_DOMAIN) &&
+      // Bucket name valid for CF
+      validR2Buckets.filter(b => l.includes(b)).length
+    )
```

Comment:
lets tighten this up to map the locations to URLs and match on the hostname and path explicitly rather than using string.includes. If the plan is to let anyone post a claim, then we have to be defensive when processing them.

Reply:
added the `account-id` in R2. However, it would actually make sense to spec how we plan to send these URLs for R2 as they vary quite a lot in R2 docs...

Reply:
I am not sure if we can actually do this nicely with S3 given bucket name is in the hostname and we can have several. I think we will need to sync on this

Reply:
see https://github.com/web3-storage/w3infra/issues/316